### ISSUE TITLE
invalid content/format of external DO should produce an error

### ISSUE THREAD

**Issue by amolari** [F5Networks/f5-bigip-runtime-init#33]
I've configured my DO declaration to be externally fetch:
```
extension_services:
  service_operations:
    - extensionType: do
      type: url
      value: https://${bigip_setup_bucket}/${tf_templatized_do_bucket_key}
```

My DO declaration was valid but in YAML format, which is incorrect/unsupported. However, the runtime-init process didn't return an error and was stuck/waiting.

Last lines in log `/var/log/cloud/bigIpRuntimeInit.log`:
```
2021-11-23T08:43:00.983Z [17945]: info: Executing service operations.
2021-11-23T08:43:00.985Z [17945]: silly: Loading file via url - options: {"locationType":"url","verifyTls":true}
```

Kept waiting for minutes, no change.

In log `/var/log/restnoded/restnoded.log` I see:
```
Tue, 23 Nov 2021 08:42:32 GMT - finest: [LoaderWorker] triggered at path:  /var/config/rest/iapps/f5-declarative-onboarding/nodejs/restWorker.js
Tue, 23 Nov 2021 08:42:32 GMT - severe: [LoaderWorker] JSHINT issues encountered in file /var/config/rest/iapps/f5-declarative-onboarding/nodejs/restWorker.js, 12 errors
Tue, 23 Nov 2021 08:42:32 GMT - severe: [LoaderWorker] /var/config/rest/iapps/f5-declarative-onboarding/nodejs/restWorker.js: line 525, col 5, Option 'validthis' can't be used in a global scope.
Tue, 23 Nov 2021 08:42:32 GMT - severe: [LoaderWorker] /var/config/rest/iapps/f5-declarative-onboarding/nodejs/restWorker.js: line 605, col 5, Option 'validthis' can't be used in a global scope.
Tue, 23 Nov 2021 08:42:32 GMT - severe: [LoaderWorker] /var/config/rest/iapps/f5-declarative-onboarding/nodejs/restWorker.js: line 632, col 5, Option 'validthis' can't be used in a global scope.
Tue, 23 Nov 2021 08:42:32 GMT - severe: [LoaderWorker] /var/config/rest/iapps/f5-declarative-onboarding/nodejs/restWorker.js: line 690, col 5, Option 'validthis' can't be used in a global scope.
Tue, 23 Nov 2021 08:42:32 GMT - severe: [LoaderWorker] /var/config/rest/iapps/f5-declarative-onboarding/nodejs/restWorker.js: line 747, col 5, Option 'validthis' can't be used in a global scope.
Tue, 23 Nov 2021 08:42:32 GMT - severe: [LoaderWorker] /var/config/rest/iapps/f5-declarative-onboarding/nodejs/restWorker.js: line 882, col 5, Option 'validthis' can't be used in a global scope.
Tue, 23 Nov 2021 08:42:32 GMT - severe: [LoaderWorker] /var/config/rest/iapps/f5-declarative-onboarding/nodejs/restWorker.js: line 921, col 5, Option 'validthis' can't be used in a global scope.
Tue, 23 Nov 2021 08:42:32 GMT - severe: [LoaderWorker] /var/config/rest/iapps/f5-declarative-onboarding/nodejs/restWorker.js: line 1011, col 5, Option 'validthis' can't be used in a global scope.
Tue, 23 Nov 2021 08:42:32 GMT - severe: [LoaderWorker] /var/config/rest/iapps/f5-declarative-onboarding/nodejs/restWorker.js: line 1123, col 5, Option 'validthis' can't be used in a global scope.
Tue, 23 Nov 2021 08:42:32 GMT - severe: [LoaderWorker] /var/config/rest/iapps/f5-declarative-onboarding/nodejs/restWorker.js: line 1181, col 5, Option 'validthis' can't be used in a global scope.
Tue, 23 Nov 2021 08:42:32 GMT - severe: [LoaderWorker] /var/config/rest/iapps/f5-declarative-onboarding/nodejs/restWorker.js: line 1222, col 5, Option 'validthis' can't be used in a global scope.
Tue, 23 Nov 2021 08:42:32 GMT - severe: [LoaderWorker] /var/config/rest/iapps/f5-declarative-onboarding/nodejs/restWorker.js: line 1323, col 5, Option 'validthis' can't be used in a global scope.
Tue, 23 Nov 2021 08:42:32 GMT - info: [f5-declarative-onboarding: restWorker.js] Running DO version: 1.25.0
Tue, 23 Nov 2021 08:42:32 GMT - info: [f5-declarative-onboarding: restWorker.js] Created Declarative onboarding worker
Tue, 23 Nov 2021 08:42:32 GMT - info: starting /shared/declarative-onboarding
Tue, 23 Nov 2021 08:42:32 GMT - info: [WorkerStateManager] Storage worker reports no key for: /shared/declarative-onboarding
Tue, 23 Nov 2021 08:42:33 GMT - info: [f5-declarative-onboarding: doUtil.js] Platform: BIG-IP
Tue, 23 Nov 2021 08:42:33 GMT - info: [f5-declarative-onboarding: restWorker.js] Initial startup
Tue, 23 Nov 2021 08:42:33 GMT - config: [RestWorker] /shared/declarative-onboarding has started. Name:RestWorker
Tue, 23 Nov 2021 08:42:33 GMT - finest: [restnodedOdataAdapter] Forwarder notify callback: {"external":"/mgmt/shared/declarative-onboarding","internal":"/shared/declarative-onboarding","port":8105,"apiStatus":"NO_STATUS"}
```

If the DO declaration fetch from an URL cannot be json validated, it should produce an error in `/var/log/cloud/bigIpRuntimeInit.log`

**Comment by shyawnkarim** [comment]
Thanks reporting this issue and providing this detailed feedback. We are now tracking this issue with internal ID ESECLDTPLT-2911.

**Comment by shyawnkarim** [comment]
Closing. 

This ticket was included with [Release 1.4.1](https://github.com/F5Networks/f5-bigip-runtime-init/releases/tag/1.4.1).
